services/contributions.js

```diff
@@ -0,0 +1,96 @@
+const githubService = require('../services/githubService')
+const tasks = require('../models/tasks')
+
+/**
+ * Get the  contributions of the user
+ * @param {string} username
+ */
+
+async function getUserContributions (username) {
+  const contributions = {}
+  const { data } = await githubService.fetchPRsByUser(username)
+  const allUserTasks = await tasks.fetchUserTasks(username)
+  const noteworthy = []
+  const all = []
+
+  if (data.total_count) {
+    const allPRsDetails = extractPRdetails(data)
+    const prMaps = new Map()
+    allPRsDetails.forEach(pr => {
+      prMaps.set(pr.url, pr)
+    })
+    allUserTasks.forEach(task => {
+      const noteworthyObject = {}
+      noteworthyObject.task = extractTaskdetails(task)
+      const prList = []
+      task.links.forEach(link => {
+        const prObject = prMaps.get(link)
+        if (prObject) {
+          prList.push(prObject)
+          prMaps.delete(link)
+        }
+      })
+      noteworthyObject.prList = prList
+
+      if (task.isNoteworthy) {
+        noteworthy.push(noteworthyObject)
+      } else {
+        all.push(noteworthyObject)
+      }
+    })
+    for (const prDetails of prMaps.values()) {
+      const allObject = {
+        prList: prDetails,
+        task: {}
+      }
+      all.push(allObject)
+    }
+  }
+  contributions.noteworthy = noteworthy
+  contributions.all = all
+  return contributions
+}
+
+/**
+ * Extracts only the necessary details required from the object returned by Github API
+ * @param data {Object} - Object returned by Github API
+ */
+
+const extractPRdetails = (data) => {
+  const allPRs = []
+  data.items.forEach(({ title, user, html_url: url, state, created_at: createdAt, updated_at: updatedAt }) => {
+    allPRs.push({
+      title,
+      username: user.login,
```

Comment:
`username` doesn't seem to be a part of `prList` object in API contract 🤔 
![image](https://user-images.githubusercontent.com/56217868/105846478-40e6ca00-6002-11eb-9446-e8066be15a89.png)